langlib/lang.array/src/main/ballerina/src/lang.array/array.bal

```diff
@@ -48,50 +66,173 @@ public function iterator(Type[] arr) returns abstract object {
+#
+# + arr - the array
+# + func - a predicate to apply to each element to determine if it should be included
+# + return - new array only containig members which evaluate function 'func' to true
 public function filter(Type[] arr, function(Type val) returns boolean func) returns Type[] = external;
 
+# Reduce operate on each member of `arr` using combining function `func` to produce
+# a new value combining all members of `arr`.
+#
+# + arr - the array
+# + func - combining function
+# + initial - initial value to first evaluation of combining function `func`
+# + return - result of applying combining function to each member of the array
+#
+# Example: Emulating sum function.
+# ```
+# var ar = [1, 2, 3];
+# var a = ar.reduce(function (int i, int j) returns int { return i + j; }, 0);
+# ```
+#
+# Example: Emulating map behavior.
+# ```
+# var ar = [1, 2, 3];
+# int[] newArr = [];
+# int[] a = ar.reduce(function (int[] a, int j) returns int[] { a.push(j*2); return a; }, newArr);
+# ```
 public function reduce(Type[] arr, function(Type1 accum, Type val) returns Type1 func, Type1 initial) returns Type1 = external;
 
-// Sub-array
-
-
-// TODO Fix this.
+// TODO: Add default arg for `endIndex`
+# Returns a sub array starting from `startIndex` (inclusive) to `endIndex` (exclusive).
+#
+# + arr - the array
+# + startIndex - index of first member to include in the slice
+# + endIndex - index of first member not to include in the slice
+# + return - array slice within specified range
 public function slice(Type[] arr, int startIndex, int endIndex) returns Type[] = external;
 
+# Removes the member of `arr` and index `i` and returns it.
+# Panics if `i` is out of range.
+#
+# + arr - the array
+# + i - index of member to be removed
+# + return - removed member
 public function remove(Type[] arr, int i) returns Type = external;
 
+# Removes all members of `arr`.
+# Panics if any member cannot be removed.
+#
+# + arr - the array
 public function removeAll((any|error)[] arr) returns () = external;
 
+# Increase or decrease the length.
+# `setLength(arr, 0)` is equivalent to `removeAll(arr)`.
+#
+# + arr - the array
+# + i - new length
 public function setLength((any|error)[] arr, int i) returns () = external;
 
+# Returns the index of first member of `arr` that is equal to `val` if there is one.
+# Returns `()` if not found
+# Equality is tested using `==`
+#
+# + arr - the array
+# + val - member to search for
+# + startIndex - index to start the search from
+# + return - index of the member if found, else `()`
 public function indexOf(PureType[] arr, PureType val, int startIndex = 0) returns int? = external;
 
+# Reverse the order of the members of `arr`.
+# Returns `arr`.
+#
+# + arr - the array to be reversed
+# + return - reversed `arr`
 public function reverse(Type[] arr) returns Type[] = external;
 
+# Sort `arr` using `func` to order members.
+# Returns `arr`.
+#
+# + arr - the array
+# + func - comparator function
```

Comment:
```suggestion
# + func - the comparator function
```

Reply:
in this context `func` is 'a' function user provide